added section to install docs for multiple cidrs

bfallonf opened the pull request:
For: https://trello.com/c/Y4Zl06Gw/469-document-make-the-cluster-ip-range-more-flexible-by-allowing-multiple-subnets-for-host-and-service-ranges-sdnoperationssupportab

@knobunc @JacobTanenbaum WDYT?

JacobTanenbaum commented on install_config/configuring_sdn.adoc:
```diff
@@ -142,7 +143,39 @@ networkConfig:
+  serviceNetworkCIDR: 172.30.0.0/16
+----
+<1> Cluster network for node IP allocation
+<2> Ensure the first `cidr` entry matches the `clusterNetworkCIDR` entry
```
the hostSubneLength of the first entry will also have to match

knobunc replied:
@JacobTanenbaum -- All the hostSubnetLengths need to match at the moment, right?

JacobTanenbaum commented on install_config/configuring_sdn.adoc:
```diff
@@ -142,7 +143,39 @@ networkConfig:
+
+Multiple ranges can be used at once, and the range can be expanded. Pods can
+also be moved from one range to another. Pod allocations occur in the order
+listed, then when the range is full, move to the next on the list.
```
Nodes are assigned to the first empty cidr range in the list. Pods are assigned to nodes by the master there is no guarantee into which cidr range a pod will be assigned.

knobunc replied:
It should be "Node allocations" not "Pod allocations" and then otherwise is fine.

knobunc commented on install_config/configuring_sdn.adoc:
```diff
@@ -142,7 +143,39 @@ networkConfig:
+Alternatively, you can create a pod network with multiple CIDR ranges by
+adding separate ranges into the `clusterNetworks` field with the range and the `hostSubnetLength`. 
+
+Multiple ranges can be used at once, and the range can be expanded. Pods can
```
range can be expanded, and if no nodes are using a range, contracted.

"Pods can be moved" => "Nodes can be moved from one range to another by draining, deleting, and re-creating the node [see link for more details]".

knobunc commented on install_config/configuring_sdn.adoc:
```diff
@@ -142,7 +143,39 @@ networkConfig:
+*redhat/openshift-ovs-multitenant* for the *ovs-multitenant* plug-in, or
+*redhat/openshift-ovs-networkpolicy* for the *ovs-networkpolicy* plug-in
+
+You can change the `clusterNetworks` value, but be sure to restart the
```
"You can add elements to the `clusterNetworks` value, or remove them if no node is using that CIDR, but be sure to restart ..."

bfallonf commented:
Thanks @JacobTanenbaum @knobunc . Changed to suggested.

@openshift/team-documentation please peer review.

bfallonf commented:
[rev_history]
|xref:../install_config/configuring_sdn.adoc#install-config-configuring-sdn[Configuring the SDN]
|Added section on configuring the SDN for multiple CIDR ranges to the xref:../install_config/configuring_sdn.adoc#configuring-the-pod-network-on-masters[Configuring the Pod Network on Masters] section.
%

ahardin-rh commented on install_config/configuring_sdn.adoc:
```diff
@@ -142,7 +143,43 @@ networkConfig:
+<1> Cluster network for node IP allocation.
+<2> Ensure the first `cidr` entry matches the `clusterNetworkCIDR` entry.
+<3> Number of bits for pod IP allocation within a node. Must match with all other `hostSubnetLength` entries in the `NetworkConfig`. 
+<4> Set to *redhat/openshift-ovs-subnet* for the *ovs-subnet* plug-in,
```
Should redhat/openshift-ovs-subnet, redhat/openshift-ovs-multitenant, and redhat/openshift-ovs-networkpolicy be in backticks?

ahardin-rh commented:
Just one question. Also, we'll need to squash this into 1 commit. 🥞

duyanyan commented on install_config/configuring_sdn.adoc:
```diff
@@ -142,7 +143,43 @@ networkConfig:
+
+You can add elements to the `clusterNetworks` value, or remove them if no node
+is using that CIDR range, but be sure to restart the `atomic-openshift-master`
+service for any changes to take effect.
```
For OCP version >= 3.7, we need to restart `atomic-openshift-master-api` and `atomic-openshift-master-controllers` to make it take effect.

bfallonf commented:
Thanks @ahardin-rh . Updated to suggestion.

@duyanyan Thanks. That's interesting. I don't think the rest of the docs are updated to reflect this. I've updated this doc, and I'll figure out what needs to happen.

I'll merge this.
